R/ML.SpeedGLMSGD.R

```diff
@@ -0,0 +1,240 @@
+#'     algorithm runs on.
+#'   }
+#'
+#'   \item{\code{get_validity}}{
+#'     Active method. Function that shows wheter the current configuration of
+#'     the initialisation is valid. The function returns \code{TRUE} if everything is
+#'     specified correctly. It will throw an error (with the error messages)
+#'     when something is misspecified. This function is automatically called
+#'     after initialization.
+#'   }
+#' }
+#' 
+#' ## TO DO implement 
+ML.SpeedGLMSGD <- R6Class("ML.SpeedGLMSGD",
+  inherit = ML.Base,
+  public =
+    list(
+      fitfunname='speedglm',
+      lmclass='speedglm',
+      initialize = function() {
+        self$get_validity
+        super$initialize()
+        
+      }
+    ),  active =
+    list(get_file_name = function() {
+         return(private$file_name)
+    }
+    
+      ),
+  private =
+      list(
+           alpha = 0.001,
+           loop = 1000,
+           file_name = file.path('output', 'coef.prn'),
+      
+      do.predict = function(X_mat,Y_vals,save_model=FALSE,m.fit=NULL) {
+        X_mat <- data.frame(X_mat)
+        Y_vals <-data.frame(Y_vals)
+        data<-cbind.data.frame(Y_vals,X_mat)
+        
+        #rename colums to standardize depending on the number of columns
+        for (i in (1:(ncol(data)))){
+          if (i == 1){
+            colnames(data)[i] <- "Y"
+          }
+          if (i>1){
+            colnames(data)[i]<-paste("x",i-1, sep = "")
+          }
+        }
+        
+        
+        if(!is.null(m.fit)){
+          result<-private$predict_lr(X_mat,m.fit)
+          
+        }
+        else {
+            try(m.fit<-private$read_model(),silent = TRUE)
+            if (!is.null(m.fit)){
+             
+            result<-private$predict_lr(X_mat,m.fit)
+        }
+          else {
+            ## fit the model
+            m.fit<- coef(private$do.fit(X_mat = X_mat, Y_vals = Y_vals, save_model = FALSE ))
+            result<-private$predict_lr(X_mat,m.fit)
+          }
+        }
+        
+        if (save_model){
+           private$save_model(model=coef(m.fit))
+        }
+        
+        return(result)
+      },
+
+      do.update = function(X_mat, Y_vals,save_model=FALSE, m.fit=NULL, ...) {
+        # speedlm uses the coefficients of the last model to update with more data.
+        X_mat <- data.frame(X_mat)
+        Y_vals <- data.frame(Y_vals)
+        data <- cbind.data.frame(Y_vals,X_mat)
+        
+        
+        #rename colums to standardize depending on the number of columns
+        for (i in (1:(ncol(data)))){
+          if (i == 1){
+            colnames(data)[i] <- "Y"
+          }
+          if (i>1){
+            colnames(data)[i]<-paste("x",i-1, sep = "")
+          }
+        }
+       
+       
+        if(!is.null(m.fit)){
+          result<-private$update_coef(X_mat,Y_vals,coef=m.fit,alpha=private$alpha,loop=private$loop)
+          }
+          else {
+            try(m.fit<-private$read_model(),silent = TRUE)
```

Comment:
I'm pretty sure this is a bad idea. A try with a silent catch if it fails? Why do we need this?